### PR TITLE
Create MemberPlan

### PR DESCRIPTION
**Pull request by roosnic1**
This PR introduces some additional fields to the user model and mutations that allow to register new members/users through the public API

⚠️ This PR merges into `f/WEP-23-member-plans`

### CODE REVIEW

File: packages/api/src/context.ts [hunk 1]
```diff
@@ -329,6 +361,56 @@ export async function contextFromRequest(
...
+          intentID: intent.intentID,
+          intentData: intent.intentData,
+          intentSecret: intent.intentSecret,
+          paymentData: intent.paymentData,
+          paymentMethodID: payment.paymentMethodID,
+          invoiceID: payment.invoiceID
+        }
+      })
+
+      if (!updatedPayment) throw new Error('Error during updating payment') // TODO: this check needs to be removed
```
Comment (0815pascal) [inline packages/api/src/context.ts:411]: Can you remove this check now or do you need to do some additional things?

**Review by 0815pascal — APPROVED** [review]
Looks good, only translations for French and German are missing

File: packages/api-db-mongodb/src/db/paymentMethod.ts [hunk 2]
```diff
@@ -64,6 +64,23 @@ export class MongoDBPaymentMethodAdapter implements DBPaymentMethodAdapter {
     return paymentMethods.map(({_id: id, ...data}) => ({id, ...data}))
   }
 
+  async getActivePaymentMethods(): Promise<PaymentMethod[]> {
+    const paymentMethods = await this.paymentMethods
+      .find({active: true})
+      .sort({createAd: -1})
```
Comment (penk85) [inline packages/api-db-mongodb/src/db/paymentMethod.ts:70]: createAt ?

File: packages/api/src/graphql/mutation.public.ts [hunk 3]
```diff
@@ -79,6 +89,109 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
...
+        if(userSession.user) throw new Error('Can not register authenticated user') // TODO: check this again
+        */
+
+        const memberPlan = await loaders.activeMemberPlansByID.load(memberPlanID)
+        if (!memberPlan) throw new NotFound('MemberPlan', memberPlanID)
+
+        const paymentMethod = await loaders.activePaymentMethodsByID.load(paymentMethodID)
+        if (!paymentMethod) throw new NotFound('PaymentMethod', paymentMethodID)
+
+        if (monthlyAmount <= memberPlan.amountPerMonthMin) throw new MonthlyAmountNotEnough()
```
Comment (penk85) [inline packages/api/src/graphql/mutation.public.ts:134]: Do you intend for monthlyAmount = memberPlan.amountPerMonthMin to throw an error? Should it not be: monthlyAmount < memberPlan.amountPerMonthMin ?

**Review by penk85 — CHANGES_REQUESTED** [review]
Just some small comments

**Review by faebebin — APPROVED** [review]

**Review by bishoymelek — APPROVED** [review]

File: packages/api/src/graphql/mutation.public.ts (hunk 3 above)
Comment (roosnic1) [reply packages/api/src/graphql/mutation.public.ts:134]: You're absolutely right. Thanks @penk85 👍🏾

**Pull request merged**